prow/tide/history/history.go

```diff
@@ -33,14 +38,84 @@ import (
 // Mock out time for unit testing.
 var now = time.Now
 
+type storageObject interface {
+	NewReader(context.Context) (io.ReadCloser, error)
+	NewWriter(context.Context) io.WriteCloser
+}
+
+type gcsStorageObject struct {
+	*storage.ObjectHandle
+}
+
+func (o gcsStorageObject) NewReader(c context.Context) (io.ReadCloser, error) {
+	return o.ObjectHandle.NewReader(c)
+}
+func (o gcsStorageObject) NewWriter(c context.Context) io.WriteCloser {
+	return o.ObjectHandle.NewWriter(c)
+}
+
 // History uses a `*recordLog` per pool to store a record of recent actions that
 // Tide has taken. Using a log per pool ensure that history is retained
 // for inactive pools even if other pools are very active.
 type History struct {
 	logs map[string]*recordLog
 	sync.Mutex
-
 	logSizeLimit int
+
+	gcsObj *storage.ObjectHandle
+}
+
+func readHistory(maxRecordsPerKey int, obj storageObject) (map[string]*recordLog, error) {
+	reader, err := obj.NewReader(context.Background())
```

Comment:
Why bother exposing the context in the interface if we're just going to use background?

Reply:
I was just copying the `ObjectHandle` function signatures exactly. Thats a good point, fixed.